pkg/runner/builtin/runner.go

```diff
@@ -283,11 +235,20 @@ func cname(job *types.Job) string {
 
 func (r *Runner) Poll(q queue.Queue) {
 	for {
```

Comment:
also, one other question, and this might require some changes to citadel (not sure). I'm wondering if the following will be an issue and how we can prevent ...

build #1 is submitted, all workers are full, goes into 10 second backoff
build #2 is submitted, worker is available, build #1 is sleeping
build #2 executes before build #1